client/extensions/zoninator/components/settings/zone/index.jsx

```diff
@@ -25,17 +26,35 @@ class Zone extends Component {
 		translate: PropTypes.func.isRequired,
 	}
 
+	state = {
+		showDeleteDialog: false,
+	}
+
+	showDeleteDialog = () => this.setState( { showDeleteDialog: true } );
+
+	hideDeleteDialog = () => this.setState( { showDeleteDialog: false } );
+
 	render() {
 		const { siteId, siteSlug, translate } = this.props;
+		const { showDeleteDialog } = this.state;
 
 		return (
 			<div>
 				<HeaderCake
 					backHref={ `${ settingsPath }/${ siteSlug }` }
-					actionButton={ <Button compact primary scary>{ translate( 'Delete' ) }</Button> } >
+					actionButton={ <Button compact primary scary onClick={ this.showDeleteDialog }>{ translate( 'Delete' ) }</Button> } >
 					{ translate( 'Edit zone' ) }
 				</HeaderCake>
 
+				{
+					showDeleteDialog &&
+					<DeleteZoneDialog
+						siteSlug={ siteSlug }
+						zoneName="[zone name]"
+						onConfirm={ noop }
```

Comment:
So this `noop` is just a placeholder for now. Is that right?

Reply:
That is correct :)